modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -2085,4 +2104,20 @@ public void finishCheckpoint() {
             }
         }
     }
+
+    /**
+     * Checks if the Checkpoint Buffer is currently close to exhaustion.
+     */
+    public boolean isCpBufferOverflowThresholdExceeded() {
```

Comment:
I suggest adding `assert started` and then the `checkpointPool` will not be `null`.